BaseClasses.py

```diff
@@ -603,6 +614,7 @@ def update_reachable_regions(self, player: int):
             if new_region in rrp:
                 bc.remove(connection)
             elif connection.can_reach(self):
+                assert not (new_region is None)
```

Comment:
should probably be `is not` for readability
adding a message to the assert is recommended, so maybe `assert new_region is not None, "invalid region"` 

message thing applies to the next two asserts as well

Reply:
`is not` is something that the Python inventors created with the short-sightedness of thinking in the ambiguous, imprecise English language.
It is more confusing for people who are not as familiar with English.
Everywhere else in Python, `not` operates on what comes after it. This is the one exception that causes confusion.
I think `is not` should not be used because it is only more readable for people that are very familiar with English.

Reply:
I agree that it's an odd one, but the `is not` is a separate operator and is used in a lot of places in AP. Should probably wait for a response from one of the core devs to make a decision (and maybe put it in the yet-to-be-written coding style guide, so it's consistent across the project).

Reply:
Can I offer "assert new_region" ?